src/Xamarin.Android.Build.Tasks/Tasks/GetJavaPlatformJar.cs

```diff
@@ -58,22 +60,39 @@ public override bool RunTask ()
 								platform = target_sdk.Value;
 
 							var min_sdk = uses_sdk.Attribute (androidNs + "minSdkVersion");
-							if (min_sdk != null && (!int.TryParse (min_sdk.Value, out int minSdkVersion) || minSdkVersion < XABuildConfig.NDKMinimumApiAvailable)) {
-								Log.LogWarningForXmlNode (
-										code:             "XA4216",
-										file:             AndroidManifest,
-										node:             min_sdk,
-										message:          Properties.Resources.XA4216_MinSdkVersion,
-										messageArgs:      new object [] { min_sdk?.Value, XABuildConfig.NDKMinimumApiAvailable }
-								);
+							if (min_sdk != null) {
+								// NOTE: $(SupportedOSPlatformVersion) will potentially be a version, but could also be an int
+								if (SupportedOSPlatformVersion.IndexOf ('.') == -1) {
+									SupportedOSPlatformVersion += ".0";
+								}
+								int supportedOsPlatformVersionAsInt = 0;
+								if (Version.TryParse (SupportedOSPlatformVersion, out var version)) {
+									supportedOsPlatformVersionAsInt = version.Major;
+								}
+
+								var failedToParseMinSdk = !int.TryParse (min_sdk.Value, out int minSdkVersion);
+
+								if (failedToParseMinSdk || minSdkVersion != supportedOsPlatformVersionAsInt) {
+									Log.LogCodedError ("XA1037", Properties.Resources.XA1037, min_sdk?.Value, SupportedOSPlatformVersion);
```

Comment:
Should this be an error or a warning, and when?

It should be an error if `minSdkVersion` is < `$(SupportedOSPlatformVersion)`.

*I think* it could be a *warning* if `minSdkVersion` is > `$(SupportedOSPlatformVersion)`.

Rephrased: what will break if `minSdkVersion` is > `$(SupportedOSPlatformVersion)`?  If nothing will break, then this is for "sanity checking" purposes, and I think a warning would be better.  If something could break, then an error seems appropriate.  (I can't think of anything that would break here, though.)

Reply:
I was following the behavior that has already been set on the macios side, where an error is surfaced on either type of mismatch. I think we could warn if `minSdkVersion` doesn't match `$(SupportedOSPlatformVersion)` in either direction if we wanted, but figured cross product consistency may be more desirable.

Reply:
I'm not convinced, but I'm not *not* convinced…

I wonder how hard it would be to look through some GitHub search results to see if this would break anybody.